kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -291,8 +305,16 @@ class TypeSpec private constructor(
   private fun constructorProperties(): Map<String, PropertySpec> {
     if (primaryConstructor == null) return emptyMap()
 
+    // Properties added after the initializer are not permitted to be inlined into the constructor
```

Comment:
Something that we can improve in future?

Reply:
Nope it's not allowed in kotlin